### PR TITLE
add role diagram

### PR DESCRIPTION
**Pull request by aaronpk**
adds a diagram illustrating the relationship between the roles in GNAP

for #217

### CODE REVIEW

File: draft-ietf-gnap-core-protocol.md [hunk 1]
```diff
@@ -130,6 +130,41 @@ The parties in GNAP perform actions under different roles.
...
+│   Server    │            │   Server   │
+│             │◄─┐   ┌────►│            │
+└─────────────┘  │   │     └────────────┘
+       ▲         │   │
+       +         │   │
+       +         │   │
+       +         │   │
+       +         ▼   ▼
+       +       ┌──────────┐
+       +       │  Client  │
```
Comment (jricher) [inline draft-ietf-gnap-core-protocol.md:147]: Do we want this to be a "client instance" instead?

**Review by jricher — CHANGES_REQUESTED** [review]
Diagram itself is OK I think, but the line/arrow style is different from the other diagrams in the doc (this one seems to use higher ASCII/ANSI line characters instead of low-ASCII?). I'm fine with either but I think it should be consistent.

**Comment by aaronpk** [comment]
Replaced with ascii chars!

**Review by jricher — APPROVED** [review]

**Comment by yaronf** [comment]
We have been using the "goat" tool in one of my drafts to convert ASCII art to SVG. The results are very nice. You may want to consider this option.

* Tool: https://github.com/blampe/goat
* Makefile: https://github.com/yaronf/I-D/blob/main/STAR-Delegation/Makefile
* Source ASCII art: https://github.com/yaronf/I-D/tree/main/STAR-Delegation/art
* Plaintext rendering: https://www.ietf.org/archive/id/draft-ietf-acme-star-delegation-09.txt (see e.g. Fig. 11)
* HTML+SVG rendering: https://www.ietf.org/archive/id/draft-ietf-acme-star-delegation-09.html

**Pull request merged**